python/pyarrow/array.pxi

```diff
@@ -1835,23 +1835,25 @@ cdef class Array(_PandasConvertible):
         This is a low-level function intended for expert users.
         """
         cdef:
-            void* c_ptr = _as_c_pointer(in_ptr)
+            ArrowDeviceArray* c_device_array = <ArrowDeviceArray*>_as_c_pointer(in_ptr)
             void* c_type_ptr
             shared_ptr[CArray] c_array
 
+        if c_device_array.device_type == 2:
```

Comment:
We could perhaps instead add `kDLCUDA` to https://github.com/apache/arrow/blob/f8784acbfcd4a2811dd404d4d53a89b26569b05a/python/pyarrow/includes/libarrow.pxd#L1322-L1323

Reply:
That's for DLPack, and also those definitions are the same, I would prefer to not mix them in our code. But added a similar line to libarrow.pxd to expose `ARROW_DEVICE_CUDA`, which also makes the above easier to read